### PR TITLE
Bridge plugin: if veth already exists - return information from corespodning ns

### PR DESCRIPTION
**Pull request by dshulyak**
addresses: https://github.com/containernetworking/cni/issues/281

### CODE REVIEW

**Comment by dshulyak** [comment]
@steveeJ what do you think about this approach? Is it something that can be asked from every cni plugin?

File: plugins/main/bridge/bridge.go [hunk 1]
```diff
@@ -170,6 +171,26 @@ func calcGatewayIP(ipn *net.IPNet) net.IP {
...
+		if err != nil {
+			return nil
+		}
+		// TODO check for v6
+		addrs, err := netlink.AddrList(link, nl.FAMILY_V4)
+		if err != nil {
+			return nil
+		}
+		result.IP4 = &types.IPConfig{
+			IP:      addrs[0].IPNet,
```
Comment (bboreham) [inline plugins/main/bridge/bridge.go]: check `len(addrs)>0` before doing this

**Comment by dshulyak** [comment]
@rosenhouse i will cover my change with tests, but what is spec change?

**Comment by rosenhouse** [comment]
@dshulyak If I'm understanding you correctly, you are proposing a change to every CNI plugin, and therefore to the CNI spec: that GET should be idempotent and return information about already-configured interfaces.

**Comment by jellonek** [comment]
@dshulyak [this spec](https://github.com/containernetworking/cni/blob/master/SPEC.md).

**Comment by dcbw** [comment]
@containernetworking/cni-maintainers to me GET is more appropriate than an idempotent ADD.  Semantically ADD implies doing the operation again, which seems odd since it was already done.  I'd rather do GET.  It seems like whatever calls CNI should really know if the container is already running, since it has to pass things like the container name, the namespace path, etc, and in that case it just wants to grab the current network config, so GET is appropriate.

**Comment by leodotcloud** [comment]
I agree with @dcbw that's it's really odd to get the same call to ADD twice. We should just return an error and the calling party should handle it. CNI plugins are invoked by container runtimes which are generally very sophisticated and have the ability to store state. I believe adding a GET is also really an overhead to the API. Other comments: https://github.com/containernetworking/cni/issues/281#issuecomment-255271661

**Comment by bboreham** [comment]
Although not fully decided on whether to add `GET` or not, we all seem agreed that `GET` would be preferable to this idempotent-`ADD` proposal, so I'm closing this one.

**Pull request closed without merging**